src/test/java/uk/gov/companieshouse/chipsrestinterfacesconsumer/producer/OutgoingMessageProducerUnitTest.java

```diff
@@ -0,0 +1,99 @@
+package uk.gov.companieshouse.chipsrestinterfacesconsumer.producer;
+
+import org.apache.kafka.clients.producer.RecordMetadata;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.jupiter.MockitoExtension;
+import uk.gov.companieshouse.chipsrestinterfacesconsumer.common.ApplicationLogger;
+import uk.gov.companieshouse.chipsrestinterfacesconsumer.model.ChipsKafkaMessage;
+import uk.gov.companieshouse.kafka.producer.CHKafkaProducer;
+import uk.gov.companieshouse.service.ServiceException;
+
+import java.io.IOException;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+public class OutgoingMessageProducerUnitTest {
+
+    private static final Future<RecordMetadata> MOCKED_FUTURE = Mockito.mock(Future.class);
```

Comment:
i wouldn't have final mocks, i would put them in a beforeEach (or whatever the new equivalent version is ), otherwise if they are used in more than one test, the usage statistics of the mock won't be cleared down between tests. It will still pass a verify in test 2 if it was called in test 1.

Reply:
or if you think they will only be needed for 1 test, just have them inside the test.

Reply:
Moved this around